--- ethtool-3.16.orig/internal.h	2014-07-21 02:21:34.000000000 +0400
+++ ethtool-3.16/internal.h	2015-03-24 20:04:21.086813504 +0300
@@ -17,16 +17,16 @@
 
 /* ethtool.h expects these to be defined by <linux/types.h> */
 #ifndef HAVE_BE_TYPES
-typedef __uint16_t __be16;
-typedef __uint32_t __be32;
+typedef uint16_t __be16;
+typedef uint32_t __be32;
 typedef unsigned long long __be64;
 #endif
 
 typedef unsigned long long u64;
-typedef __uint32_t u32;
-typedef __uint16_t u16;
-typedef __uint8_t u8;
-typedef __int32_t s32;
+typedef uint32_t u32;
+typedef uint16_t u16;
+typedef uint8_t u8;
+typedef int32_t s32;
 
 #include "ethtool-copy.h"
 #include "net_tstamp-copy.h"
