diff --git a/jconfig.cfg b/jconfig.cfg
index 36a04fa..c01e806 100644
--- a/jconfig.cfg
+++ b/jconfig.cfg
@@ -7,8 +7,6 @@
 #undef void
 #undef const
 #undef CHAR_IS_UNSIGNED
-#undef HAVE_STDDEF_H
-#undef HAVE_STDLIB_H
 #undef NEED_BSD_STRINGS
 #undef NEED_SYS_TYPES_H
 #undef NEED_FAR_POINTERS
@@ -18,6 +16,8 @@
 
 #ifdef JPEG_INTERNALS
 
+#undef HAVE_STDDEF_H
+#undef HAVE_STDLIB_H
 #undef RIGHT_SHIFT_IS_UNSIGNED
 #undef INLINE
 /* These are for configuring the JPEG memory manager. */
