diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-11-12 12:56:21.186707497 +0300
+++ b/CMakeLists.txt	2020-11-12 11:25:22.281535000 +0300
@@ -100,6 +100,8 @@
 if (ENABLE_SYSTEMD)
   target_link_libraries(stubby PRIVATE Libsystemd::Libsystemd)
 endif()
+target_link_libraries(stubby PRIVATE crypto)
+target_link_libraries(stubby PRIVATE ssl)
 # Are we being built from getdns? If so, use the build tree getdns.
 if (TARGET getdns)
   target_link_libraries(stubby PRIVATE getdns)
