diff --git a/coregrind/m_syswrap/syswrap-mips32-linux.c b/coregrind/m_syswrap/syswrap-mips32-linux.c
index c70bc21..ae05467 100644
--- a/coregrind/m_syswrap/syswrap-mips32-linux.c
+++ b/coregrind/m_syswrap/syswrap-mips32-linux.c
@@ -411,6 +411,7 @@ DECL_TEMPLATE (mips_linux, sys_pipe);
 DECL_TEMPLATE (mips_linux, sys_prctl);
 DECL_TEMPLATE (mips_linux, sys_ptrace);
 DECL_TEMPLATE (mips_linux, sys_sync_file_range);
+DECL_TEMPLATE (mips_linux, sys_setdomainname);
 
 PRE(sys_mmap2) 
 {
@@ -747,6 +748,12 @@ PRE(sys_sync_file_range)
    }
 }
 
+PRE(sys_setdomainname)
+{
+   PRINT ("sys_setdomainname ( %#lx, %ld )", ARG1, SARG2);
+   PRE_REG_READ2 (long, "setdomainname", const void *, name, int, len);
+}
+
 #undef PRE
 #undef POST
 
@@ -886,15 +893,15 @@ static SyscallTableEntry syscall_main_table[] = {
    GENX_ (__NR_fsync,                  sys_fsync),                   // 118
    PLAX_ (__NR_sigreturn,              sys_sigreturn),               // 119
    LINX_ (__NR_clone,                  sys_clone),                   // 120
-   //..    //   (__NR_setdomainname,     sys_setdomainname),     // 121
+   PLAX_ (__NR_setdomainname,          sys_setdomainname),           // 121
    GENXY (__NR_uname,                  sys_newuname),                // 122
    //..    PLAX_(__NR_modify_ldt,        sys_modify_ldt),        // 123
    LINXY (__NR_adjtimex,               sys_adjtimex),                // 124
    GENXY (__NR_mprotect,               sys_mprotect),                // 125
    LINXY (__NR_sigprocmask,            sys_sigprocmask),             // 126
    //..    GENX_(__NR_create_module,     sys_ni_syscall),        // 127
-   //..    GENX_(__NR_init_module,       sys_init_module),       // 128
-   //..    //   (__NR_delete_module,     sys_delete_module),     // 129
+   LINX_ (__NR_init_module,            sys_init_module),             // 128
+   LINX_ (__NR_delete_module,          sys_delete_module),           // 129
    //..    GENX_(__NR_get_kernel_syms,   sys_ni_syscall),        // 130
    //..    LINX_(__NR_quotactl,          sys_quotactl),          // 131
    GENX_ (__NR_getpgid,                sys_getpgid),                 // 132
