diff --git a/coregrind/m_debuginfo/lzodefs.h b/coregrind/m_debuginfo/lzodefs.h
index 0e40e33..274c041 100644
--- a/coregrind/m_debuginfo/lzodefs.h
+++ b/coregrind/m_debuginfo/lzodefs.h
@@ -44,6 +44,8 @@
 #ifndef __LZODEFS_H_INCLUDED
 #define __LZODEFS_H_INCLUDED 1
 
+#include <features.h>
+
 #if defined(__CYGWIN32__) && !defined(__CYGWIN__)
 #  define __CYGWIN__ __CYGWIN32__
 #endif
diff --git a/coregrind/m_debuginfo/minilzo-inl.c b/coregrind/m_debuginfo/minilzo-inl.c
index 153d008..dc1969c 100644
--- a/coregrind/m_debuginfo/minilzo-inl.c
+++ b/coregrind/m_debuginfo/minilzo-inl.c
@@ -63,6 +63,7 @@
 #endif
 #include <limits.h>
 #include <stddef.h>
+#include <features.h>
 #if defined(MINILZO_CFG_USE_INTERNAL_LZODEFS)
 
 #ifndef __LZODEFS_H_INCLUDED
diff --git a/include/pub_tool_basics.h b/include/pub_tool_basics.h
index d1119e1..2f25e28 100644
--- a/include/pub_tool_basics.h
+++ b/include/pub_tool_basics.h
@@ -46,7 +46,7 @@
 
 // For varargs types
 #include <stdarg.h>
-
+#include <features.h>
 
 /* ---------------------------------------------------------------------
    symbol prefixing
