diff -Nuar argp-standalone-1.3/argp-help.c argp-standalone-1.3-patched/argp-help.c
--- argp-standalone-1.3/argp-help.c	2022-09-08 15:46:45.619416703 +0300
+++ argp-standalone-1.3-patched/argp-help.c	2022-09-08 15:47:08.979517510 +0300
@@ -1002,6 +1002,9 @@
 filter_doc (const char *doc, int key, const struct argp *argp,
 	    const struct argp_state *state)
 {
+  if (! argp)
+    return doc;
+
   if (argp->help_filter)
     /* We must apply a user filter to this output.  */
     {
