diff -Naur ./Configurations/10-main.conf ./Configurations/10-main.conf
--- ./Configurations/10-main.conf	2019-09-30 17:38:26.656504341 +0300
+++ ./Configurations/10-main.conf	2019-09-30 17:37:02.138916000 +0300
@@ -638,10 +638,10 @@
         CXX              => "g++",
         CFLAGS           => picker(default => "-Wall",
                                    debug   => "-O0 -g",
-                                   release => "-O3"),
+                                   release => ""),
         CXXFLAGS         => picker(default => "-Wall",
                                    debug   => "-O0 -g",
-                                   release => "-O3"),
+                                   release => ""),
         cflags           => threads("-pthread"),
         cxxflags         => combine("-std=c++11", threads("-pthread")),
         lib_cppflags     => "-DOPENSSL_USE_NODELETE",
