diff -Nuar gdb-8.2.1/gdb/linux-nat.c gdb-8.2.1-patched/gdb/linux-nat.c
--- gdb-8.2.1/gdb/linux-nat.c	2020-01-17 16:04:57.979953822 +0300
+++ gdb-8.2.1-patched/gdb/linux-nat.c	2020-01-17 16:05:08.604254190 +0300
@@ -469,6 +469,7 @@
       /* Detach new forked process?  */
       if (detach_fork)
 	{
+	  int detach = 1;
 	  struct cleanup *old_chain = make_cleanup (delete_lwp_cleanup,
 						    child_lp);
 
@@ -493,9 +494,11 @@
 		perror_with_name (_("Couldn't do single step"));
 	      if (my_waitpid (child_pid, &status, 0) < 0)
 		perror_with_name (_("Couldn't wait vfork process"));
+	      else
+		detach = WIFSTOPPED (status);
 	    }
 
-	  if (WIFSTOPPED (status))
+	  if (detach)
 	    {
 	      int signo;
 
