diff -Nuar gdb-8.2.1/readline/util.c gdb-8.2.1-patched/readline/util.c
--- gdb-8.2.1/readline/util.c	2017-06-04 18:51:27.000000000 +0300
+++ gdb-8.2.1-patched/readline/util.c	2020-01-17 17:20:45.168560257 +0300
@@ -517,7 +517,7 @@
 	    : "."),
 	   getpid());
 #else
-  sprintf (fnbuf, "/var/tmp/rltrace.%ld", getpid());
+  sprintf (fnbuf, "/var/tmp/rltrace.%ld", (long)getpid());
 #endif
   unlink(fnbuf);
   _rl_tracefp = fopen (fnbuf, "w+");
