diff -Nuar autoconf-2.69/lib/autoconf/c.m4 autoconf-2.69-patched/lib/autoconf/c.m4
--- autoconf-2.69/lib/autoconf/c.m4	2012-01-21 17:46:39.000000000 +0400
+++ autoconf-2.69-patched/lib/autoconf/c.m4	2017-11-26 22:35:27.036480539 +0300
@@ -578,7 +578,7 @@
 # Make sure it works both with $CC and with simple cc.
 # We do the test twice because some compilers refuse to overwrite an
 # existing .o file with -o, though they will create one.
-ac_try='$CC -c conftest.$ac_ext -o conftest2.$ac_objext >&AS_MESSAGE_LOG_FD'
+ac_try='$CC $CFLAGS -c conftest.$ac_ext -o conftest2.$ac_objext >&AS_MESSAGE_LOG_FD'
 rm -f conftest2.*
 if _AC_DO_VAR(ac_try) &&
    test -f conftest2.$ac_objext && _AC_DO_VAR(ac_try);
@@ -586,8 +586,8 @@
   eval ac_cv_prog_cc_${ac_cc}_c_o=yes
   if test "x$CC" != xcc; then
     # Test first that cc exists at all.
-    if _AC_DO_TOKENS(cc -c conftest.$ac_ext >&AS_MESSAGE_LOG_FD); then
-      ac_try='cc -c conftest.$ac_ext -o conftest2.$ac_objext >&AS_MESSAGE_LOG_FD'
+    if _AC_DO_TOKENS(cc $CFLAGS -c conftest.$ac_ext >&AS_MESSAGE_LOG_FD); then
+      ac_try='cc $CFLAGS -c conftest.$ac_ext -o conftest2.$ac_objext >&AS_MESSAGE_LOG_FD'
       rm -f conftest2.*
       if _AC_DO_VAR(ac_try) &&
 	 test -f conftest2.$ac_objext && _AC_DO_VAR(ac_try);
